lib/elixir/lib/module/types/descr.ex

```diff
@@ -2378,15 +2380,17 @@ defmodule Module.Types.Descr do
       acc ->
         try do
           {tag, fields} = map_literal_intersection(tag1, pos1, tag2, pos2)
-          entry = {tag, fields, negs1 ++ negs2}
+          negs = negs1 ++ (negs2 -- negs1)
+          entry = {tag, fields, negs}
 
           # Imagine a, b, c, where a is closed and b and c are open with
           # no keys in common. The result in both cases will be a and we
           # want to avoid adding duplicates, especially as intersection
           # is a cartesian product.
-          case :lists.member(entry, acc) do
-            true -> acc
-            false -> [entry | acc]
+          cond do
+            :lists.member({tag, fields}, negs) -> acc
```

Comment:
Can we remove these checks? I think the will be much harder to do once we change the representation of maps. We should keep only the `:lists.member(entry, acc)` for now. Do you think that would be enough?